geode-redis/src/main/java/org/apache/geode/redis/internal/executor/key/RenameNxExecutor.java

```diff
@@ -0,0 +1,32 @@
+package org.apache.geode.redis.internal.executor.key;
+
+import static org.apache.geode.redis.internal.RedisConstants.ERROR_NO_SUCH_KEY;
+
+import java.util.List;
+
+import org.apache.geode.redis.internal.data.ByteArrayWrapper;
+import org.apache.geode.redis.internal.executor.AbstractExecutor;
+import org.apache.geode.redis.internal.executor.RedisResponse;
+import org.apache.geode.redis.internal.netty.Command;
+import org.apache.geode.redis.internal.netty.ExecutionHandlerContext;
+
+public class RenameNxExecutor extends AbstractExecutor {
+  @Override
+  public RedisResponse executeCommand(Command command, ExecutionHandlerContext context) {
+
+    List<ByteArrayWrapper> commandElems = command.getProcessedCommandWrappers();
+    ByteArrayWrapper key = command.getKey();
+    ByteArrayWrapper newKey = commandElems.get(2);
+    RedisKeyCommands redisKeyCommands = getRedisKeyCommands(context);
+
+    if (redisKeyCommands.exists(newKey)) {
+      return RedisResponse.integer(0);
+    }
+
+    if (!redisKeyCommands.rename(key, newKey)) {
```

Comment:
I think this will need a bit more work since there is a race condition here. Consider the case when two separate requests (on different servers) perform the check above and each request makes it through the check...

Before digging into that, I'd suggest writing a concurrent test to highlight the problem here.

Reply:
will do, thx.